internal/cmd/module/list.go

```diff
@@ -1,112 +1,93 @@
 package module
 
 import (
+	"context"
 	"fmt"
 	"slices"
 
 	"github.com/pkg/errors"
 	"github.com/shurcooL/graphql"
 	"github.com/spacelift-io/spacectl/client/structs"
+	"github.com/spacelift-io/spacectl/internal"
 	"github.com/spacelift-io/spacectl/internal/cmd"
 	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
 	"github.com/urfave/cli/v2"
 )
 
-const (
-	maxSearchModulesPageSize = 50
-)
-
 func listModules() cli.ActionFunc {
 	return func(cliCtx *cli.Context) error {
 		outputFormat, err := cmd.GetOutputFormat(cliCtx)
 		if err != nil {
 			return err
 		}
 
+		var limit *uint
+		if cliCtx.IsSet(cmd.FlagLimit.Name) {
+			limit = internal.Ptr(cliCtx.Uint(cmd.FlagLimit.Name))
+		}
+
+		var search *string
+		if cliCtx.IsSet(cmd.FlagSearch.Name) {
+			search = internal.Ptr(cliCtx.String(cmd.FlagSearch.Name))
+		}
+
 		switch outputFormat {
 		case cmd.OutputFormatTable:
-			m, err := getModules(cliCtx, 20)
-			if err != nil {
-				return err
-			}
-			return listModulesTable(m)
+			return listModulesTable(cliCtx, search, limit)
```

Comment:
Should we set a default limit on the table output, such as 10 items, and display the limit below the table?

Reply:
It will now default to 20 as it did before.

Reply:
Could you also print the limit at the end of table output, so that the user knows they are limited? I think the original confusion came from the fact that we started to limit, but we did not tell. Otherwise, not having the limit is better.

Reply:
<img width="584" alt="image" src="https://github.com/user-attachments/assets/333eaa43-d016-4b0b-a54a-2444b5adfbb3" />